src/app.js

```diff
@@ -81,8 +81,16 @@ export function app(app) {
             }).data
           )
 
-          if (result != null && result.then == null) {
-            repaint((state = merge(state, emit("update", result))))
+          function update(data) {
+            if (data != null) {
+              repaint((state = merge(state, emit("update", data))))
+            }
+          }
+
+          if (result != null && typeof result.then == "function") {
+            result.then(update)
+          } else {
+            update(result)
           }
```

Comment:
Let's change this block to:

```js
          if (result != null && result.then != null) {
            result.then(update)
          } else {
            update(result)
          }
```

and move the `update` implementation before the implementation of `repaint` right after `return emit`, around like 35.

Rewrite it like this please.

```js
  function update(_state) {
    if (_state != null) {
      repaint((state = merge(state, emit("update", _state))))
    }
  }
```